src/openrct2/world/MapHelpers.cpp

```diff
@@ -240,11 +242,14 @@ int32_t tile_smooth(int32_t x, int32_t y)
 
             // Get neighbour height. If the element is not valid (outside of map) assume the same height
             auto* neighbourSurfaceElement = map_get_surface_element_at(x + x_offset, y + y_offset);
-            neighbourHeightOffset.baseheight[index] = neighbourSurfaceElement ? neighbourSurfaceElement->base_height
-                                                                              : surfaceElement->base_height;
+            if (neighbourSurfaceElement != nullptr)
```

Comment:
Not required the ternary already checks for if the element is nullptr. Perhaps make the ternary more explicit by adding an explicit comparison with nullptr